contracts/modules/staking/StakingProducts.sol

```diff
@@ -597,10 +607,87 @@ contract StakingProducts is IStakingProducts, MasterAwareV2, Multicall {
     return surgePremium / allocationUnitsPerNxm;
   }
 
+  /* ========== STAKING POOL CREATION ========== */
+
+  function stakingPool(uint poolId) public view returns (IStakingPool) {
+    return IStakingPool(
+      StakingPoolLibrary.getAddress(address(stakingPoolFactory), poolId)
+    );
+  }
+
+  function createStakingPool(
+    bool isPrivatePool,
+    uint initialPoolFee,
+    uint maxPoolFee,
+    ProductInitializationParams[] memory productInitParams,
+    string calldata ipfsDescriptionHash
+  ) external whenNotPaused onlyMember returns (uint /*poolId*/, address /*stakingPoolAddress*/) {
+
+    uint numProducts = productInitParams.length;
+
+    // override with initial price and check if pool is allowed
+    for (uint i = 0; i < numProducts; i++) {
+
+      if (productInitParams[i].targetPrice < GLOBAL_MIN_PRICE_RATIO) {
+        revert TargetPriceBelowGlobalMinPriceRatio();
+      }
+
+      uint productId = productInitParams[i].productId;
+
+      ICoverProducts _coverProducts = coverProducts();
+      // if there is a list of allowed pools for this product - this pool didn't exist yet so it's not in it
+      if (_coverProducts.allowedPoolsCount(productId) > 0) {
+        revert PoolNotAllowedForThisProduct(productId);
+      }
+
+      if (productId >= _coverProducts.productsCount()) {
+        revert ProductDoesntExistOrIsDeprecated();
+      }
+
+      Product memory product = _coverProducts.products(productId);
+
+      if (product.isDeprecated) {
+        revert ProductDoesntExistOrIsDeprecated();
+      }
+
+      productInitParams[i].initialPrice = product.initialPriceRatio;
+    }
+
+    (uint poolId, address stakingPoolAddress) = IStakingPoolFactory(stakingPoolFactory).create(coverContract);
+
+    IStakingPool(stakingPoolAddress).initialize(
+      isPrivatePool,
+      initialPoolFee,
+      maxPoolFee,
+      poolId,
+      ipfsDescriptionHash
+    );
+
+    tokenController().assignStakingPoolManager(poolId, msg.sender);
+
+    _setInitialProducts(poolId, productInitParams);
+
+    return (poolId, stakingPoolAddress);
+  }
+
   /* dependencies */
 
+  function tokenController() internal view returns (ITokenController) {
```

Comment:
Dep only used in createStakingPool, remove after moving that one out.

Reply:
pending decision to keep or move out